File: 프로그래머스/2/64065. 튜플/튜플.py
def solution(s):
    answer = []
    s = s[2:-2]
    s = s.split("},{")
    s.sort(key = len)
    
    for nums in s:
        #print(nums)
        numss = nums.split(",")
        for n in numss:
            if int(n) not in answer:
                answer.append(int(n))
    
    return answer

# 길이 순으로 소팅해서 없는 값만 가지고 오기 

File: 프로그래머스/2/64065. 튜플/튜플.java
import java.util.*;

class Solution {
    public int[] solution(String s) {
        s = s.substring(2, s.length());
        s = s.substring(0, s.length()-2).replace("},{", "-");
        String str[] = s.split("-");
        Arrays.sort(str, (a, b) -> (a.length()-b.length()));
        
        List<Integer> list = new ArrayList<>();
        
        for (String st : str) {
            String[] temp = st.split(",");
            for (String c :temp) {
                int n = Integer.parseInt(c);
                if (!list.contains(n)) {
                    list.add(n);
                    break;
                }
            }
        }
        int[] answer = new int[list.size()];
        for ( int i = 0; i< list.size() ; i++)  {
            answer[i] = list.get(i);
        }
        
        return answer;
    }
}